app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt

```diff
@@ -401,10 +401,14 @@ class RefreshController @Inject constructor(
         return impactedThreads
     }
 
+    private fun List<Int>.getAtStart(size: Int) = subList(0, size) to subList(size, count())
+
+    private fun List<Int>.getAtEnd(size: Int) = subList(count() - size, count()) to subList(0, count() - size)
+
     private fun computeUids(folder: Folder, direction: Direction): Pair<List<Int>, List<Int>> {
         val allUids = if (direction == Direction.TO_THE_FUTURE) folder.newMessagesUidsToFetch else folder.oldMessagesUidsToFetch
         return if (allUids.count() > Utils.PAGE_SIZE) {
-            allUids.subList(0, Utils.PAGE_SIZE) to allUids.subList(Utils.PAGE_SIZE, allUids.count())
+            if (direction == Direction.TO_THE_FUTURE) allUids.getAtEnd(Utils.PAGE_SIZE) else allUids.getAtStart(Utils.PAGE_SIZE)
```

Comment:
We're certain that `allUids` field will always be filled in this very exact order? With oldest messages at the start and newest message at the end?